crypto/s2n_pkey_evp.c

```diff
@@ -182,7 +182,10 @@ int s2n_pkey_evp_sign(const struct s2n_pkey *priv, s2n_signature_algorithm sig_a
     DEFER_CLEANUP(EVP_PKEY_CTX *pctx = s2n_evp_pkey_ctx_new(priv->pkey, hash_state->alg), EVP_PKEY_CTX_free_pointer);
     POSIX_ENSURE_REF(pctx);
     POSIX_GUARD_OSSL(EVP_PKEY_sign_init(pctx), S2N_ERR_PKEY_CTX_INIT);
-    POSIX_GUARD_OSSL(S2N_EVP_PKEY_CTX_set_signature_md(pctx, s2n_hash_alg_to_evp_md(hash_state->alg)), S2N_ERR_PKEY_CTX_INIT);
+
+    if (sig_alg != S2N_SIGNATURE_MLDSA) {
+        POSIX_GUARD_OSSL(S2N_EVP_PKEY_CTX_set_signature_md(pctx, s2n_hash_alg_to_evp_md(hash_state->alg)), S2N_ERR_PKEY_CTX_INIT);
+    }
```

Comment:
How is the libcrypto informed of the signature algorithm in the case of `S2N_SIGNATURE_MLDSA`?

Reply:
How is it informed of the signature algorithm in the case of other cert types? ;) Note that we're setting the hash algorithm here, not the signature algorithm. "md" stands for "message digest". ML-DSA doesn't have a "message digest" like this. The actual signature algorithm comes from the EVP_PKEY itself.